API/Controllers/TournamentsController.cs

```diff
@@ -1,34 +1,223 @@
+    [HttpPatch("{id:int}")]
+    [Authorize(Roles = "admin")]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType<ModelStateDictionary>(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType<TournamentDTO>(StatusCodes.Status200OK)]
+    public async Task<IActionResult> UpdateAsync(
+        int id,
+        [FromBody] JsonPatchDocument<TournamentDTO> patch
+    )
+    {
+        // Ensure target tournament exists
+        TournamentDTO? tournament = await _tournamentsService.GetAsync(id);
+        if (tournament is null)
+        {
+            return NotFound();
+        }
+
+        // Ensure request is only attempting to perform a replace operation.
+        if (!patch.IsReplaceOnly())
+        {
+            return BadRequest();
+        }
+
+        // Patch and validate
+        patch.ApplyTo(tournament, ModelState);
+        if (!TryValidateModel(tournament))
+        {
+            return BadRequest(ModelState);
+        }
+
+        // Apply patched values to entity
+        TournamentDTO? updatedTournament = await _tournamentsService.UpdateAsync(id, tournament);
+        return Ok(updatedTournament!);
+    }
+
+    /// <summary>
+    /// Submit matches to an existing tournament
+    /// </summary>
+    /// <param name="id">Tournament id</param>
+    /// <param name="wrapper">Match submission data</param>
+    /// <param name="verify">Optionally verify all included matches</param>
+    /// <response code="401">If verify is true and the User does not have match verification privileges</response>
+    /// <response code="404">If a tournament matching the given id does not exist</response>
+    /// <response code="201">Returns location information of the created matches</response>
+    [HttpPost("{id:int}/matches")]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType<IEnumerable<MatchCreatedResultDTO>>(StatusCodes.Status201Created)]
+    public async Task<IActionResult> CreateMatchesAsync(
+        int id,
+        [FromBody] MatchesWebSubmissionDTO wrapper,
+        [FromQuery] bool verify = false
+    )
+    {
+        MatchVerificationSource? verificationSource = null;
+        if (verify)
+        {
+            if (!User.CanVerifyMatches())
+            {
+                return Unauthorized();
+            }
+
+            verificationSource = User.VerificationSource();
+        }
+
+        IEnumerable<MatchCreatedResultDTO>? result =
+            await _matchesService.CreateAsync(
+                id,
+                wrapper.SubmitterId,
+                wrapper.Ids, verify,
+                (int?)verificationSource
+            );
+
+        if (result is null)
+        {
+            return NotFound();
+        }
+
+        // Use no location header for multiple creation
+        return Created((string?)null, result);
```

Comment:
What's going on with the `(string?)null`?

Reply:
This was done by the formatter I guess? I will late a look as to why